File: Problem-3/largestPrimeFactor.py
#Anton Ilic, Feb 25, 2023
#https://projecteuler.net/problem=3
#Find the largest prime factor of 600851475143
#very inneficiant and slow, but will eventually give the right answer 

def isPrime(number):
    if number < 2: #-ive, 0, 1 return false.
        return False 
    for i in range(2, number // 2 + 1):
        if(number%i == 0):
            return False
    return True

#Taking from the bottom
def largestPrimeFactorFinder(number):
    largestPrimeFactor = 1
    for x in range(2, number // 2 + 1):
        if(number%x == 0):
            if(isPrime(x)):
                largestPrimeFactor = x
        x += 1
    return largestPrimeFactor

number = 600851475143 
print(largestPrimeFactorFinder(number))

File: Problem-3/largestPrimeFactor.java

public class largestPrimeFactor {
    public static void main(String[] args) {
        long number =  600851475143L;
        largestPrimeFactor factor = new largestPrimeFactor();
        long largestPrimeFactor = factor.findLargestPrimeFactor(number);
        System.out.println(largestPrimeFactor);
    }

    /**
     * Finds the largest prime factor of the number. 
     * 
     * @param number Number which we wish to find the largest Prime factor of. 
     * @return largest Prime Factor
     */
    private long findLargestPrimeFactor(long number){
        long i = 2;
        while(i * i < number){
            while(number % i == 0){
                number = number / i;
            }
            i = i + 1;
        }
        return number;
    }

}
